lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -149,13 +178,36 @@ class RenderBlockingResources extends Audit {
    * @param {Node} fcpGraph
    * @param {Set<string>} deferredIds
    * @param {Map<string, number>} wastedCssBytesByUrl
+   * @param {LH.Artifacts.DetectedStack[]} Stacks
    * @return {number}
    */
-  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl) {
-    const originalEstimate = simulator.simulate(fcpGraph).timeInMs;
+  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl, Stacks) {
+    const {nodeTimings} = simulator.simulate(fcpGraph);
+    const stackSpecificTimings = new Map(nodeTimings);
 
     let totalChildNetworkBytes = 0;
     const minimalFCPGraph = /** @type {NetworkNode} */ (fcpGraph.cloneWithRelationships(node => {
+      if (nodeTimings) {
+        const nodeTiming = nodeTimings.get(node);
```

Comment:
nit: extract this block to another function? the logic inside this clone loop is getting quite nested and large at this point